google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurationPanel.java

```diff
@@ -145,25 +141,32 @@ public void applyEditorTo(@NotNull AppEngineDeploymentConfiguration configuratio
    */
   public void setDeploymentProjectAndVersion(DeploymentSource deploymentSource) {
     if (deploymentSource instanceof AppEngineDeployable) {
-      ((AppEngineDeployable) deploymentSource).setProjectName(projectSelector.getText());
+      ((AppEngineDeployable) deploymentSource)
+          .setProjectName(
+              Optional.ofNullable(projectSelector.getSelectedProject())
+                  .map(CloudProject::projectId)
+                  .orElse(""));
       ((AppEngineDeployable) deploymentSource)
           .setVersion(
               Strings.isNullOrEmpty(versionIdField.getText()) ? "auto" : versionIdField.getText());
     }
   }
 
   /**
-   * Updates the text of the panel as follows: if the project text box is empty, no message is
-   * displayed, if the project text represents a valid project, the project details are displayed,
-   * if the project text represents an invalid project, an error message is displayed.
+   * Updates the text of the panel as follows: if no project is selected, no message is displayed,
+   * if the project represents a valid project, the project details are displayed, if the project
+   * represents an invalid project, an error message is displayed.
    */
-  private void refreshApplicationInfoPanel() {
-    if (Strings.isNullOrEmpty(projectSelector.getText())) {
+  private void refreshApplicationInfoPanel(CloudProject selectedProject) {
```

Comment:
why pass it in now instead of pulling it from the ProjectSelector as before?

Reply:
With updated API project updated event is an updated `CloudProject` so it's easy to just pass it in as a parameter instead of calling it directly. It might be useful for more unit testing later as well. What do you think, does it feel it's less readable?